accounting/invoice/utils.py

```diff
@@ -21,27 +21,102 @@
 Invoice utilities.
 """
 
-from datetime import timedelta
+import datetime
 
-from django.utils import timezone
+from django.conf import settings
+from django.core.mail.message import EmailMessage
+from django.template.loader import get_template
 
+from accounting.account.models import Account
+from accounting.invoice import choices, models
 
-def get_last_day_past_month():
+
+def resolve_dict_callables(dictionary, *args):
     """
-    Return the `datetime` object representing the last day of the month that just passed.
+    Resolve any potential callables in a dictionary by passing it some arguments.
```

Comment:
@UmanShahzad This would benefit from a concrete example :) You could tailor the example to the use case below (where `message_context` is updated with additional data from the result of doing `resolve_dict_callables(extra_email_context, invoice)`).

Reply:
Sure, I can give it an example :+1: